specutils/spectra/spectrum_mixin.py

```diff
@@ -84,10 +84,7 @@ def spectral_axis_unit(self):
         """
         Returns the units of the spectral axis.
         """
-        if isinstance(self.wcs, HighLevelWCSWrapper):
-            return u.Unit(self.wcs.world_axis_units[0])
-
-        return self.wcs.unit[0]
+        return u.Unit(self.wcs.world_axis_units[0])
```

Comment:
```suggestion
        if hasattr(self.wcs, 'world_axis_units'):
            return u.Unit(self.wcs.world_axis_units[0])
        else:
            return u.Unit(self.wcs.unit[0])
```

Reply:
I'm not sure about this. The `self.wcs.unit[0]` formalism is the pre-astropy 4.0 method specifically for FITS WCSs. But since the current specutils claims to be incompatible with anything less that astropy 4.0, and we have made it such that it can work with anything that follows the APE14 WCS paradigm, I'm really hesitant to build in fail safes that suggest we aren't really committed to that.

Reply:
Point taken. Especially given #582.  Already, I retract my suggestion then.